src/main/com/fulcrologic/fulcro/rendering/ident_optimized_render.cljc

```diff
@@ -17,13 +17,11 @@
   "Checks the given `idents` and returns a subset of them where the data they refer to has changed
    between `old-state` and `new-state`."
   [old-state new-state idents]
-  (reduce
-    (fn [result ident]
-      (if (identical? (get-in old-state ident) (get-in new-state ident))
-        result
-        (cons ident result)))
-    (list)
-    idents))
```

Comment:
This was written with lists and cons *for* speed. Cons is making a linked list, and adding to the head is very very fast. There is no intermediate data here, and your impl is probably slower.

Reply:
You might be right. I was tempted to replace it `(list)` with `#{}` but that would mean changing the API.

Reply:
I just did a manual test:

```
(time (do 
        (reduce (fn [r v]
                 (cons v r)) (list) (range 1 100000))
        :ok))
"Elapsed time: 1.98474 msecs"
```

vs yours:

```
(time (do 
        (into [] (range 1 100000))
        :ok))
"Elapsed time: 2.073323 msecs"
```

Notice that the per-item insertion time for both is around 20ns each...and I'd say the usualy number of idents on-screen in an app might average about 100-1000...so 20 microseconds.

Premature optimization for sure...and it is not really any different.